### PR TITLE
Add validator for changes in the set of struct field attributes that …

### PR DESCRIPTION
**Pull request by geirst**
…are part of complex fields.

Currently, adding attribute aspect requires refeed.

@toregge and @baldersheim please review

### CODE REVIEW

File: config-model/src/main/java/com/yahoo/searchdefinition/derived/AttributeFields.java [hunk 1]
```diff
@@ -153,6 +154,12 @@ public Iterator attributeIterator() {
         return Collections.unmodifiableCollection(attributes.values());
     }
 
+    public Collection<Attribute> structFieldAttributes(String baseFieldName) {
+        return attributes().stream()
+                .filter(attribute -> attribute.getName().startsWith(baseFieldName + "."))
```
Comment (baldersheim) [inline config-model/src/main/java/com/yahoo/searchdefinition/derived/AttributeFields.java]: Even though your mother will probably optimize this one for you I consider it a good habit to do
String structPrefix = baseFieldName + ".";
and then use that in the filter

**Review by baldersheim — APPROVED** [review]
Apart from minor comment it looks good to me.
However I think that and hands on go through tomorrow morning to ensure I get the overall rules is a good idea.

File: config-model/src/main/java/com/yahoo/searchdefinition/derived/AttributeFields.java (hunk 1 above)
Comment (geirst) [reply config-model/src/main/java/com/yahoo/searchdefinition/derived/AttributeFields.java]: Fixed

**Comment by baldersheim** [comment]
Merge at will.

**Review by toregge — APPROVED** [review]
LGTM

**Pull request merged**